var/spack/repos/builtin/packages/coreneuron/package.py

```diff
@@ -126,6 +126,10 @@ def get_flags(self):
             flags = '-xHost -qopt-report=5'
             if '+knl' in spec:
                 flags = '-xMIC-AVX512 -qopt-report=5'
+        # NVHPC 21.11 detects ABM support and defines __ABM__, which breaks
+        # Random123 compilation
+        if spec.satisfies('%nvhpc@21.11'):
+            flags += ' -mno-abm'
```

Comment:
did you post the nv-forum issue? as we will have neuron update in upstream Spack, this is something we have to remember as todo.

Reply:
The suggestion of `-mno-abm` came from https://forums.developer.nvidia.com/t/21-11-tp-behaviour/198115.
My conclusion is that it's a Random123 bug: it assumes that `intrin.h` will be present on systems defining `__ABM__`, and BB5 is a counterexample. In the medium term this should be fixed in Random123, or worked around from CMake in CoreNEURON and NEURON.

Reply:
https://github.com/BlueBrain/CoreNeuron/issues/724